src/main/scala/mercury/PageScanner.scala

```diff
@@ -34,11 +34,17 @@ object PageScanner {
         .find(_.hasAttr("data-component"))
         .map(_.attr("data-component"))
 
-    val links = for (link <- elems) yield {
-      val href = link.attr("abs:href").takeWhile('?' != _).takeWhile('#' != _)
-      val comp = findDataComponent(link)
-      val isSublink = link.parents().asScala.exists(e => e.attr("data-link-name").contains("sublink"))
-      SimpleLink(href, comp, isSublink)
+    val links = elems.flatMap { link =>
+
+      val component = findDataComponent(link).filterNot {
+        c => c == "most-popular" || c.startsWith("popular-in-")
+      }
+
+      component.map { c =>
+        val href = link.attr("abs:href").takeWhile('?' != _).takeWhile('#' != _)
```

Comment:
It looks like you're writing half a URL parser.  Have you considered https://github.com/smola/galimatias, which we use in Ophan?